### PR TITLE
CLOUDP-67424: implement atlas security customerCerts describe

### PR DESCRIPTION
**Pull request by robcarlan-mlab**
## Proposed changes

_Jira ticket:_ [mongocli atlas security certs list|ls](https://jira.mongodb.org/browse/CLOUDP-67424)

Closes #[issue number]

## Checklist

- [] I have signed the [MongoDB CLA](https://www.mongodb.com/legal/contributor-agreement)
- [] I have added tests that prove my fix is effective or that my feature works
- [] I have added any necessary documentation (if appropriate)
- [] I have run `make fmt` and formatted my code

## Further comments

```
./bin/mongocli atlas security customerCerts describe --projectId 5cac6a2ba6f239298a18a025
{
  "cas": "-----BEGIN CERTIFICATE-----\nredacted\n-----END CERTIFICATE-----\n"
}
```

### CODE REVIEW

File: internal/cli/atlas/security/customercerts/certs.go [hunk 1]
```diff
@@ -0,0 +1,30 @@
...
+package customercerts
+
+import (
+	"github.com/mongodb/mongocli/internal/description"
+	"github.com/spf13/cobra"
+)
+
+func Builder() *cobra.Command {
+	cmd := &cobra.Command{
+		Use:   "customerCerts",
```
Comment (gssbzn) [inline internal/cli/atlas/security/customercerts/certs.go:24]: [nit] let's add an alias for the singular, and also for all lowercase (customercerts| customercert)

File: internal/cli/atlas/security/customercerts/describe_test.go [hunk 2]
```diff
@@ -0,0 +1,49 @@
...
+
+// +build unit
+
+package customercerts
+
+import (
+	"testing"
+
+	"go.mongodb.org/atlas/mongodbatlas"
+
```
Comment (gssbzn) [inline internal/cli/atlas/security/customercerts/describe_test.go:23]: [nit] extra empty line

File: internal/description/description.go [hunk 3]
```diff
@@ -126,7 +128,7 @@ const (
 	CreateOwner              = "Create the first user for Ops Manager."
 	Servers                  = "Manage Ops Manager servers."
 	ListServer               = "List all available servers running an automation agent for the given project."
-	Security                 = "Manage clusters security configuration."
+	Security                 = "Manage security configuration for a project."
```
Comment (gssbzn) [inline internal/description/description.go:131]: 👍

File: internal/description/description.go [hunk 4]
```diff
@@ -45,6 +45,8 @@ const (
 	CreateDataLake           = "Create a new data lake."
 	UpdateDataLake           = "Update a data lake."
 	DeleteDataLake           = "Delete a data lake."
+	Certs                    = "Manage x509 certificates for your database users."
```
Comment (gssbzn) [inline internal/description/description.go:48]:
```suggestion
	Certs                    = "Customer managed x509 certificates for your database users."
```

File: internal/cli/atlas/security/customercerts/describe.go [hunk 5]
```diff
@@ -0,0 +1,68 @@
...
+	"github.com/mongodb/mongocli/internal/json"
+	"github.com/mongodb/mongocli/internal/store"
+	"github.com/mongodb/mongocli/internal/usage"
+	"github.com/spf13/cobra"
+)
+
+type DescribeOpts struct {
+	cli.GlobalOpts
+	cli.ListOpts
+	store store.X509CertificateStore
```
Comment (gssbzn) [inline internal/cli/atlas/security/customercerts/describe.go:31]: [nit]
```suggestion
	store store. X509CertificateDescriber
```

File: internal/store/x509_auth_database_users.go [hunk 6]
```diff
@@ -0,0 +1,44 @@
...
+
+type X509CertificateDescriber interface {
+	X509Configuration(string) (*atlas.CustomerX509, error)
+}
+
+type X509CertificateStore interface {
+	X509CertificateDescriber
+}
+
+// X509Certificates retrieves the current user managed certificates for a database user
```
Comment (gssbzn) [inline internal/store/x509_auth_database_users.go:35]:
```suggestion
// X509Configuration retrieves the current user managed certificates for a database user
```

File: internal/description/description.go (hunk 4 above)
Comment (robcarlan-mlab) [reply internal/description/description.go:48]: The rest of the descriptions all list the actions the command can do, does "Manage your own x509 certificates ..." work? Or "Manage customer x509 certificates"

Comment (gssbzn) [reply internal/description/description.go]: >Manage customer x509 certificates

Works for me, just trying to clarify this, doesn't matter if the other commands mention is customer related as `mongocli security customerCerts --help` will only show this help text

File: internal/cli/atlas/security/customercerts/certs.go [hunk 7]
```diff
@@ -0,0 +1,31 @@
...
+
+import (
+	"github.com/mongodb/mongocli/internal/description"
+	"github.com/spf13/cobra"
+)
+
+func Builder() *cobra.Command {
+	cmd := &cobra.Command{
+		Use:     "customerCerts",
+		Aliases: []string{"certs", "customercerts"},
```
[the remark this answers is not in the document]
Comment (robcarlan-mlab) [reply internal/cli/atlas/security/customercerts/certs.go:25]: Oh sorry yeah missed that in the original comment. yep agreed

**Pull request merged**